### PR TITLE
[posix] add pseudo-reset

### PR DESCRIPTION
**Pull request by FiveDimensions**
Add posix pseudo reset

### CODE REVIEW

File: src/posix/platform/system.cpp [hunk 1]
```diff
@@ -42,12 +42,18 @@
...
 
 otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
 {
     otInstance *instance = NULL;
 
+    if (gPlatformPseudoResetWasRequested)
+    {
+        gPlatformPseudoResetWasRequested = false;
+    }
+
```
Comment (simonlingoogle) [inline src/posix/platform/system.cpp:56]:
```suggestion
    gPlatformPseudoResetWasRequested = false;

```

File: src/posix/main.c [hunk 2]
```diff
@@ -392,7 +385,7 @@ int main(int argc, char *argv[])
 #endif
 #endif
 
-    while (true)
+    while (!otSysPseudoResetWasRequested())
```
Comment (bukepo) [inline src/posix/main.c:388]: can we make otSysMainloopPoll() return `-2` to indicate requesting pseudo-reset? Thus we don't need a new API.

Comment (FiveDimensions) [reply src/posix/main.c:388]: ok

**Comment by bukepo** [comment]
ping~

**Comment by simonlingoogle** [comment]
> ping~

Could any one answer why we need pseudo-reset on posix ?

**Comment by jwhui** [comment]
@FiveDimensions , any plans to move this PR forward?

**Pull request closed without merging**